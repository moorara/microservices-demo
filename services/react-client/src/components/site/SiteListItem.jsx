import React from 'react'
import PropTypes from 'prop-types'
import { Link } from 'react-router-dom'

// Presentational Component
const SiteListItem = ({ site }) => (
  <tr>
    <td><Link to={`/sites/${site.id}`}>{site.name}</Link></td>
    <td>{site.location}</td>
    <td>{site.priority}</td>
    <td>{site.tags.join(' ')}</td>
  </tr>
)

SiteListItem.propTypes = {
  site: PropTypes.object.isRequired
}

export default SiteListItem
