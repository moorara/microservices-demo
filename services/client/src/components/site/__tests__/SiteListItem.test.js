import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'

import SiteListItem from '../SiteListItem'

describe('SiteListItem', () => {
  const props = {
    site: {}
  }

  test('renders react component with props', () => {
    props.site = { id: '1111-1111', name: 'Station 1', location: 'Toronto, ON', priority: 2, tags: [ 'oxygen', 'hydrogen' ] }
    const wrapper = shallow(<SiteListItem {...props} />)

    expect(shallowToJson(wrapper)).toMatchSnapshot()
    expect(wrapper.find(`Link [to="/sites/${props.site.id}"]`))
    expect(wrapper.find('td').at(1).text()).toBe(`${props.site.location}`)
    expect(wrapper.find('td').at(2).text()).toBe(`${props.site.priority}`)
    expect(wrapper.find('td').at(3).text()).toBe(`${props.site.tags.join(' ')}`)
  })
  test('renders react component with props', () => {
    props.site = { id: '2222-2222', name: 'Station 2', location: 'Kingston, ON', priority: 3, tags: [ 'gold', 'silver' ] }
    const wrapper = shallow(<SiteListItem {...props} />)

    expect(shallowToJson(wrapper)).toMatchSnapshot()
    expect(wrapper.find(`Link [to="/sites/${props.site.id}"]`))
    expect(wrapper.find('td').at(1).text()).toBe(`${props.site.location}`)
    expect(wrapper.find('td').at(2).text()).toBe(`${props.site.priority}`)
    expect(wrapper.find('td').at(3).text()).toBe(`${props.site.tags.join(' ')}`)
  })
})
