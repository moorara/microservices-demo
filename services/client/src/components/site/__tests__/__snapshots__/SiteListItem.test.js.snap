// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SiteListItem renders react component with props 1`] = `
<tr>
  <td>
    <Link
      to="/sites/1111-1111"
    >
      Station 1
    </Link>
  </td>
  <td>
    Toronto, ON
  </td>
  <td>
    2
  </td>
  <td>
    oxygen hydrogen
  </td>
</tr>
`;

exports[`SiteListItem renders react component with props 2`] = `
<tr>
  <td>
    <Link
      to="/sites/2222-2222"
    >
      Station 2
    </Link>
  </td>
  <td>
    Kingston, ON
  </td>
  <td>
    3
  </td>
  <td>
    gold silver
  </td>
</tr>
`;
